fieldservice_sale/models/sale_order_line.py

```diff
@@ -31,6 +33,26 @@ def _compute_product_updatable(self):
             else:
                 super(SaleOrderLine, line)._compute_product_updatable()
 
+    @api.multi
+    @api.depends('product_id')
+    def _compute_qty_delivered_method(self):
+        super(SaleOrderLine, self)._compute_qty_delivered_method()
+        for line in self:
+            if not line.is_expense and line.product_id.type == 'service' \
+               and line.product_id.field_service_tracking == 'order':
+                line.qty_delivered_method = 'field'
+
+    @api.multi
+    @api.depends('fsm_order_id.stage_id')
+    def _compute_qty_delivered(self):
+        super(SaleOrderLine, self)._compute_qty_delivered()
+        lines_by_fsm = self.filtered(lambda sol:
+                                     sol.qty_delivered_method == 'field')
+        complete = self.env.ref('fieldservice.fsm_stage_completed')
+        for line in lines_by_fsm:
+            if line.fsm_order_id.stage_id == complete:
```

Comment:
initialize qty with 0 otherwise, if you change state, qty will still retain the old value.